src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -172,6 +172,105 @@ impl FromStr for PolyOverZq {
     }
 }
 
+impl<IntoZq: Into<Zq>> From<IntoZq> for PolyOverZq {
+    /// Create a constant [`PolyOverZq`] with a specified constant.
+    ///
+    /// # Parameters:
+    /// - `value`: the constant value the polynomial will have.
+    ///   It has to be a [`Zq`], or a value that can be converted into [`Zq`].
+    ///   
+    /// Returns a new constant polynomial with the specified value and modulus.
+    ///
+    /// # Examples:
+    /// ```
+    /// use qfall_math::{integer_mod_q::*, traits::*};
+    ///
+    /// let poly = PolyOverZq::from((1,10));
+    /// let poly_2 = PolyOverZq::from(Zq::from((1,10)));
+    ///
+    /// let value_cmp: Zq = poly.get_coeff(0).unwrap();
+    /// assert_eq!(value_cmp, Zq::from((1, 10)));
+    /// assert_eq!(poly.get_degree(), 0);
+    /// assert_eq!(poly, poly_2);
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if the provided value can not be converted into a [`Zq`].
+    ///   For example, if the modulus is not larger than one.
+    fn from(value: IntoZq) -> Self {
+        let value: Zq = value.into();
+        let poly_z = PolyOverZ::from(value.value);
+
+        PolyOverZq::from((&poly_z, &value.modulus))
```

Comment:
provide ownership for the poly_z if it is later changed to a more efficient variant

Reply:
PolyOverZq::from is not yet implemented for the ownership variant.
Therefore, I left it how it is at the moment.